src/components/XAxis.js

```diff
@@ -10,10 +10,16 @@ const XAxis = ({
   const [range0, range1, k] = [range[0] + 0.5, range[range.length - 1] + 0.5, 1]
   const domain = `M${range0},${k * tickSizeOuter}V0.5H${range1}V${k * tickSizeOuter}`
 
-  const format = scale.tickFormat ? scale.tickFormat.apply(scale) : x => x
-  const values = scale.ticks(tickCt)
+  const format = scale.tickFormat ? scale.tickFormat.apply(scale) : x => String(x)
+  const values = scale.ticks ? scale.ticks(tickCt) : scale.domain()
+
   const ticks = values.map((v, i) => {
-    const pos = scale(v)
+    let pos = scale(v)
+
+    if (typeof scale.bandwidth === 'function') {
```

Comment:
I want the ticks to be in the middle of the bands for the `<BarChart />` component. For the `<TimeSeries />` component, the scale won't have a `.bandwidth()` method so this conditional checks to see the features of the scale and then infers how to position the ticks from there.